server/lib/chat/chat-api.ts

```diff
@@ -439,4 +463,182 @@ export class AdminChatApi {
+          name,
+          limited: channelInfos.length > 0,
+          availableIn: channelInfos.map(c => c.id),
+          imagePath,
+        },
+        client,
+      )
+
+      const buffer = await image.toBuffer()
+      await writeFile(imagePath, buffer, {
+        acl: 'public-read',
+        type: imageExtension === 'png' ? 'image/png' : 'image/jpeg',
+      })
+
+      return {
+        channelBanner: toChannelBannerJson(channelBanner),
+      }
+    })
+  }
+
+  @httpPatch('/banners/:bannerId')
+  @httpBefore(checkAllPermissions('manageChannelContent'), handleMultipartFiles(MAX_IMAGE_SIZE))
+  async editChannelBanner(ctx: RouterContext): Promise<AdminEditChannelBannerResponse> {
+    const {
+      params: { bannerId },
+    } = validateRequest(ctx, {
+      params: Joi.object<{ bannerId: ChannelBannerId }>({
+        bannerId: Joi.string().uuid().required(),
+      }),
+    })
+    const originalChannelBanner = await adminGetChannelBanner(bannerId)
+
+    if (!originalChannelBanner) {
+      throw new ChatServiceError(
+        ChatServiceErrorCode.ChannelBannerNotFound,
+        'channel banner not found',
+      )
+    }
+
+    const {
+      body: { name, availableIn, deleteLimited },
+    } = validateRequest(ctx, {
+      body: Joi.object<AdminEditChannelBannerRequest & { image: any }>({
+        name: Joi.string(),
+        availableIn: Joi.array().items(Joi.string()),
+        deleteLimited: Joi.boolean(),
+        image: Joi.any(),
+      }),
+    })
+
+    const imageFile = ctx.request.files?.image
+    if (Array.isArray(imageFile)) {
+      throw new httpErrors.BadRequest('only one channel banner file can be uploaded')
+    }
+    const [image, imageExtension] = imageFile
+      ? await resizeImage(imageFile.filepath, CHANNEL_BANNER_WIDTH, CHANNEL_BANNER_HEIGHT)
+      : [undefined, undefined]
+
+    return await transact(async client => {
+      const updatedChannelBanner: Patch<Omit<ChannelBanner, 'id' | 'uploadedAt' | 'updatedAt'>> = {}
+
+      if (name) {
+        updatedChannelBanner.name = name
+      }
+      if (availableIn) {
+        const channelInfos = availableIn.length > 0 ? await findChannelsByName(availableIn) : []
+        updatedChannelBanner.limited = channelInfos.length > 0
+        updatedChannelBanner.availableIn = channelInfos.map(c => c.id)
+      }
+      if (deleteLimited) {
+        updatedChannelBanner.limited = false
+        updatedChannelBanner.availableIn = []
+      }
+      let imagePath: string | undefined
+      if (image) {
+        imagePath = createImagePath('channel-banner-images', imageExtension)
+        updatedChannelBanner.imagePath = imagePath
+      }
+
+      const channelBanner = await adminUpdateChannelBanner(bannerId, updatedChannelBanner, client)
+
+      if (image && imagePath) {
+        const buffer = await image.toBuffer()
+        await writeFile(imagePath, buffer, {
+          acl: 'public-read',
+          type: imageExtension === 'png' ? 'image/png' : 'image/jpeg',
```

Comment:
I don't think this logic is particularly great here given that you've pulled the logic for choosing conversion types away from all of this, a safer thing to do would be to use some library to choose a mimetype based on extension (or figure out if some part of this process already does that, or pull this behavior into the `writeFile` stuff with an optional override)